From 9b9c74d2787426d23b45bcc85b7c36f12836e575 Mon Sep 17 00:00:00 2001
From: "Dharageswari.R" <dharageswari.r@intel.com>
Date: Tue, 23 Jun 2015 16:06:13 +0530
Subject: [PATCH] ASoC: core: Modify snd_soc_dapm_new_pcm() to include
 "dai_link"

Update the dai_link details in the snd_soc_dapm_widget structure.

Change-Id: I0aaa283e9b5fc1c08394852631e18352308ec6b6
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-12531
Signed-off-by: Dharageswari.R <dharageswari.r@intel.com>
Signed-off-by: Samreen Nilofer <samreen.nilofer@intel.com>
---
 include/sound/soc-dapm.h | 3 ++-
 sound/soc/soc-core.c     | 4 ++--
 sound/soc/soc-dapm.c     | 7 +++++--
 3 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/include/sound/soc-dapm.h b/include/sound/soc-dapm.h
index 98781c91860f..802e90cfc031 100644
--- a/include/sound/soc-dapm.h
+++ b/include/sound/soc-dapm.h
@@ -102,7 +102,7 @@ int snd_soc_dapm_new_dai_widgets(struct snd_soc_dapm_context *dapm,
 int snd_soc_dapm_link_dai_widgets(struct snd_soc_card *card);
 void snd_soc_dapm_connect_dai_link_widgets(struct snd_soc_card *card);
 int snd_soc_dapm_new_pcm(struct snd_soc_card *card,
-			 const struct snd_soc_pcm_stream *params,
+			 struct snd_soc_dai_link *dai_link,
 			 struct snd_soc_dapm_widget *source,
 			 struct snd_soc_dapm_widget *sink);
 
@@ -229,6 +229,7 @@ struct snd_soc_dapm_widget {
 	struct snd_soc_platform *platform;
 	struct list_head list;
 	struct snd_soc_dapm_context *dapm;
+	struct snd_soc_dai_link *dai_link;
 
 	int pvt_data_len;			/* Len of widget specific data*/
 	void *priv;				/* widget specific data */
diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index 989761096292..f136a8380d72 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -1481,7 +1481,7 @@ static int soc_probe_link_dais(struct snd_soc_card *card, int num, int order)
 			}
 
 			if (play_w && capture_w) {
-				ret = snd_soc_dapm_new_pcm(card, dai_link->params,
+				ret = snd_soc_dapm_new_pcm(card, dai_link,
 						   capture_w, play_w);
 				if (ret != 0) {
 					dev_err(card->dev, "ASoC: Can't link %s to %s: %d\n",
@@ -1499,7 +1499,7 @@ static int soc_probe_link_dais(struct snd_soc_card *card, int num, int order)
 			}
 
 			if (play_w && capture_w) {
-				ret = snd_soc_dapm_new_pcm(card, dai_link->params,
+				ret = snd_soc_dapm_new_pcm(card, dai_link,
 						   capture_w, play_w);
 				if (ret != 0) {
 					dev_err(card->dev, "ASoC: Can't link %s to %s: %d\n",
diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index caa16d2ef8cb..b407af7c8920 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -3590,7 +3590,7 @@ out:
 }
 
 int snd_soc_dapm_new_pcm(struct snd_soc_card *card,
-			 const struct snd_soc_pcm_stream *params,
+			 struct snd_soc_dai_link *dai_link,
 			 struct snd_soc_dapm_widget *source,
 			 struct snd_soc_dapm_widget *sink)
 {
@@ -3623,7 +3623,10 @@ int snd_soc_dapm_new_pcm(struct snd_soc_card *card,
 		return -ENOMEM;
 	}
 
-	w->params = params;
+	w->dai_link = dai_link;
+
+	if (dai_link->params)
+		w->params = dai_link->params;
 
 	memset(&routes, 0, sizeof(routes));
 
-- 
1.9.1

